tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs

```diff
@@ -97,20 +119,72 @@ impl<BinanceClient: BinanceApi, SolanaClient: SolanaClientTrait>
 		.await?;
 
 		// TODO: this should be abstracted away
-		let (payout_amount, payout_amount_u256) = self
-			.do_binance_swap(
-				omni_account,
-				swap_order.from_asset.clone(),
-				from_amount,
-				from_address,
-				pumpx_config.wallet_index,
-				false,
-			)
-			.await?;
+		// todo we only transfer to binance without swap
+
+		let should_wait_for_deposit_confirm = false;
+
+		self.do_binance_deposit(
+			omni_account,
+			swap_order.from_asset.clone(),
+			from_amount_decimal,
+			from_address.clone(),
+			pumpx_config.wallet_index,
+			should_wait_for_deposit_confirm,
+		)
+		.await?;
+
+		if instant {
+			// we should wait for deposit confirmation in separate thread to not block intent execution and release lock upon it
+			let amount_to_release = AmountType::from_str(&from_amount.to_string()).unwrap();
+			self.account_asset_lock.release(account_id.clone(), swap_order.from_asset.clone(), amount_to_lock
+				).map_err(|e| 	log::error!("Could not release asset lock for account: {:?}, amount: {:?}, asset: {:?}, reason: {:?}", account_id, swap_order.from_asset, amount_to_release, e))?;
+		}
 
-		self.do_payout(&payout_address, &payout_amount_u256).await?;
+		let mut payout_amount =
+			self.estimate_payout(swap_order.from_asset.clone(), from_amount_decimal).await?;
+		let mut payout_amount_u256 = str_to_u256(&payout_amount, 18)?;
+
+		// trade only if deposit is confirmed
+		if should_wait_for_deposit_confirm {
```

Comment:
Is this correct? It looks like we'll never do the binance trade

Reply:
This is the current flow:
https://github.com/litentry/heima/blob/dev/tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs#L314

In my opinion it makes sense -> you simply have to wait for deposit confirmation before making any trade.

But i'm not sure whether it's correct - i tried to keep it unchanged.

Reply:
I was referring to how we set the value a few the lines above:
```rust
// TODO: this should be abstracted away
// todo we only transfer to binance without swap

let should_wait_for_deposit_confirm = false;
```
We never change this value so we'll never actually do the binance trade. Is it a placeholder? I don't remember this in the original implementation

Reply:
It's hardcoded to `false` on `dev` branch 
https://github.com/litentry/heima/blob/c95859a569531175a04d1f18c6d21cc9477ba41a/tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs#L107

Reply:
I see, I guess is some sort of placeholder then

Reply:
I'll keep it as it is.